mu-trees/addon/module-registries/requirejs.js

```diff
@@ -43,51 +43,55 @@ export default class RequireJSRegistry {
     return segments;
   }
 
-  _detectModule(specifierString, lookupMethod) {
-    let specifier = deserializeSpecifier(specifierString);
-
+  _detectModule(specifier, lookupDefault, lookupNamed) {
     let segments = this._baseSegments(specifier);
     let basePath = `${segments.join('/')}`;
     let typedPath = `${basePath}/${specifier.type}`;
 
-    let lookupResult = lookupMethod(typedPath);
+    let lookupResult = lookupDefault(typedPath);
 
-    if (
-      !lookupResult &&
-      this._config.collections[specifier.collection].defaultType === specifier.type
-    ) {
-      lookupResult = lookupMethod(basePath);
+    if (!lookupResult) {
+      if (this._checkDefaultType(specifier)) {
+        lookupResult = lookupDefault(basePath);
```

Comment:
@dgeb I'd like your feedback around here. If we don't find the default export on the basePath when a type is defaultType in this collection, should we look for a named export?